node_modules/oae-transfer/lib/internal/dao.js

```diff
@@ -0,0 +1,778 @@
+ * @api private
+ */
+ var _getFoldersByGroupIds = function (ctx, idResource, callback) {
+    Cassandra.runQuery('SELECT * FROM "FoldersGroupId" WHERE "groupId" = ?', [idResource], function(err, result) {
+        
+        // The resource is a groupe if there is no result to this query
+        if(_.chain(result).map(Cassandra.rowToHash)._wrapped[0] === undefined) {
+            GroupAPI.getGroup(ctx, idResource, function(err, group) {
+                if(err) {
+                    return callback(err);
+                }
+                return callback(1, group);
+            });
+        } else {
+            FoldersDAO.getFolder(_.chain(result).map(Cassandra.rowToHash)._wrapped[0].folderId, function(err, folder) {
+                if(err) {
+                    return callback(err);
+                }
+                return callback(null, folder);
+            });
+        }
+    });
+};
+
+/**
+ * Get the life timetransfer.
+ *
+ * @param  {Object}         ctx                     Context to determine the life time of a transfer code
+ * @param  {Function}       callback                Standard callback function
+ * @param  {Object}         callback.err            Error
+ */
+var _getLifeTimeTransfer = function (ctx, callback) {
+    if (_.isEmpty(ctx)) {
+        return callback(null, []);
+    }
+
+    var defaultTime = 1;
+    var configkey = "oae-transfer/lifetime/value";
+
+    Cassandra.runQuery('SELECT "value" FROM "Config" WHERE "tenantAlias" = ? AND "configKey" = ?', [ctx.tenant().alias, configkey], function(err, result) {
+        if (err) {
+            return callback(err);
+        }
+
+        if (_.isEmpty(result)) {
+            return callback(null, defaultTime);
+            
+        } else {
+            conf = _.chain(result).map(Cassandra.rowToHash)._wrapped[0];
+            return callback(null, conf.value);
+        }
+    });
+};
+
+/**
+ * Insert an element to the library.
+ *
+ * @param  {Object}         obj                     The element to add       
+ * @param  {Object}         principalId             The id of userTarget
+ * @param  {Function}       callback                Standard callback function
+ * @param  {Object}         callback.err            Error
+ */
+var _insertOnLibrary = function(obj, principalId, constant, callback) {
+    var entries = [];
+    entries.push({
+        'id': principalId,
+        'rank': obj.lastModified,
+        'resource': obj
+    });
+    LibraryAPI.Index.insert(constant, entries, callback);
+};
+
+/**
+ * Remove an element to the library.
+ *
+ * @param  {Object}         obj                     The element to add       
+ * @param  {Object}         principalId             The id of userTarget
+ * @param  {Function}       callback                Standard callback function
+ * @param  {Object}         callback.err            Error
+ */
+var _removeFromLibrary = function(obj, principalId, constant, callback) {
+    var entries = [];
+    entries.push({
```

Comment:
You can add this element while declaring `entries` instead of needing to push it separately.